packages/build-utils/src/fs/stream-to-buffer.ts

```diff
@@ -26,3 +26,37 @@ export default function streamToBuffer(
     });
   });
 }
+
+export function streamToBufferChunks(
+  stream: NodeJS.ReadableStream,
+  mbLimit = 100
+): Promise<Buffer[]> {
+  return new Promise<Buffer[]>((resolve, reject) => {
+    const byteLimit = 1024 * 1024 * mbLimit;
+
+    const buffers: Buffer[] = [];
+
+    let currentBuffer: Buffer;
+    stream.on('data', (chunk: Buffer) => {
+      if (!currentBuffer) {
+        currentBuffer = chunk;
+      } else if (currentBuffer.length > byteLimit) {
```

Comment:
If I'm reading correctly, the buffers in the array won't have a length of _exactly_ `byteLimit`?

Reply:
Yeah, I had 2 other approaches to getting this to exactly 100MB but then figured it probably wasn't worth the effort. Do you think it'd matter? Have a straightforward approach in mind?